practice-two/src/components/FormValidation/FormValidation.jsx

```diff
@@ -105,37 +100,37 @@ const FormValidation = ({
     }
   }, [selectedCustomer]);
 
+  useEffect(() => {
+    const isFormEmpty = Object.values(formData).some((value) => value === '');
+    setIsFormValid(!isFormEmpty);
+  }, [formData]);
+
   // Submit form
-  const handleSubmit = useCallback(
-    async (e) => {
-      e.preventDefault();
+  const handleSubmit = useCallback(async (e) => {
+    e.preventDefault();
 
-      // Check validation form when submiting
-      const newErrors = {};
-      for (const field in formData) {
-        const errorMessage = Validation(field, formData[field]);
-        if (errorMessage) {
-          newErrors[field] = errorMessage;
-        }
-      }
-      setErrors(newErrors);
+    const type = selectedCustomer ? 'UPDATE' : 'ADD';
+    let response = null;
+    let toastMessage = '';
+    if (selectedCustomer) {
+      response = await updateCustomerService(selectedCustomer.id, formData);
+    } else {
+      response = await createCustomerService(formData);
+    }
 
-      if (Object.keys(newErrors).length === 0) {
-        const response = await createCustomerService(formData);
-        let toastMessage = MESSAGES.GET.SUCCESSES.ADD_SUCCESSED;
-        if (response.error) {
-          toastMessage = MESSAGES.GET.ERRORS.ADD_FAILED;
-        } else {
-          dispatch(actionReducerCustomer(ACTION_TYPES.CREATE, response.data));
-        }
-        resetForm();
-        // To close form
-        handleToggleForm();
-        showToastInfo(toastMessage);
-      }
-    },
-    [formData, setErrors]
-  );
+    const { data, error } = response;
+    if (error) {
+      toastMessage = MESSAGES[type].FAIL;
+    } else {
+      toastMessage = MESSAGES[type].SUCCESS;
+      dispatch(actionReducerCustomer(ACTION_TYPES[type], data));
+      selectedCustomer && setSelectedCustomer(data)
+    }
+
+    resetForm();
+    onToggleForm(); // To close form
```

Comment:
should close form before reset form value

Reply:
I reversed their positions
![image](https://github.com/datdragon777/React-Training/assets/74806826/b08e31ec-aec0-42a4-980c-addc637f8711)